src/prisma/generator/templates/types.py.jinja

```diff
@@ -479,16 +479,20 @@ class {{ model.name }}CreateWithoutRelationsInput({{ model.name }}OptionalCreate
         {% endif %}
     {% endfor %}
 
+class {{ model.name }}ConnectOrCreateWithoutRelationsInput(TypedDict):
+    create: '{{ model.name }}CreateWithoutRelationsInput'
+    where: '{{ model.name }}WhereUniqueInput'
 
 class {{ model.name }}CreateNestedWithoutRelationsInput(TypedDict, total=False):
     create: '{{ model.name }}CreateWithoutRelationsInput'
     connect: '{{ model.name }}WhereUniqueInput'
+    connectOrCreate: '{{ model.name }}ConnectOrCreateWithoutRelationsInput'
```

Comment:
nit: I think this should be snake case, `connect_or_create`. You'll need to update the `QUERY_BUILDER_ALIASES` mapping in `src/prisma/_constants.py` to rename it back to `connectOrCreate` internally